### PR TITLE
3.15.x release backports

### PR DESCRIPTION
**Pull request by hjdivad**
backport #6960 #6985

### CODE REVIEW

File: packages/-ember-data/addon/index.js [hunk 1]
```diff
@@ -60,6 +58,15 @@ import JSONSerializer from '@ember-data/serializer/json';
 import RESTSerializer, { EmbeddedRecordsMixin } from '@ember-data/serializer/rest';
 
 import Model, { attr, belongsTo, hasMany } from '@ember-data/model';
+import Transform from '@ember-data/serializer/transform';
+
+if (VERSION.match(/^1\.([0-9]|1[0-2])\./)) {
```
Comment (runspired) [inline packages/-ember-data/addon/index.js]: Odd that this snuck back in.

File: packages/unpublished-relationship-performance-test-app/app/routes/application.js [hunk 2]
```diff
@@ -19,4 +19,22 @@ export default Route.extend({
       return flattened;
     });
   },
+  afterModel() {
```
Comment (runspired) [inline packages/unpublished-relationship-performance-test-app/app/routes/application.js]: surprising we picked this up but seems fine it makes the perf check more backwards compatible

**Comment by runspired** [comment]
The LTS failure here is due to ember-beta (because the current bug would affect all known versions of ember-data)

File: packages/unpublished-relationship-performance-test-app/app/routes/application.js (hunk 2 above)
Comment (hjdivad) [reply packages/unpublished-relationship-performance-test-app/app/routes/application.js]: https://github.com/emberjs/data/pull/6985/files#diff-3f38ec8323b0ede1b00d2042d47df6f3

File: packages/-ember-data/addon/index.js (hunk 1 above)
Comment (hjdivad) [reply packages/-ember-data/addon/index.js]: killed it

**Pull request merged**